library/alloc/src/macros.rs

```diff
@@ -101,9 +101,18 @@ macro_rules! vec {
 /// ```
 #[macro_export]
 #[stable(feature = "rust1", since = "1.0.0")]
+#[allow_internal_unstable(fmt_as_str)]
 macro_rules! format {
     ($($arg:tt)*) => {{
-        let res = $crate::fmt::format($crate::__export::format_args!($($arg)*));
-        res
+        let r = match $crate::__export::format_args!($($arg)*) {
+            args => {
```

Comment:
What is the point of this single-arm `match`? Just do `let args = $crate::__export::format_args!` instead.

Reply:
With `let` binding, I got borrowing error: lifetime of `&[ArgumentsV1]` in `format_args!` not live long enough.

Reply:
`format_args!` borrows from temporaries. This means that `match` is necessary to extend the lifetime of the temporaries until after the `Arguments` is no longer used. The same pattern is used inside `format_args!` itself. https://github.com/rust-lang/rust/blob/2fe9a33659641d062c1fe3577327147b4d9943a2/src/librustc_builtin_macros/format.rs#L780-L806

Reply:
Oh, what a weird pattern. Maybe add a brief comment. I didn't see this pattern before.